pytest.ini

```
@@ -4,3 +4,7 @@ doctest_optionflags = ALLOW_UNICODE ELLIPSIS
 junit_suite_name = molecule_test_suite
 norecursedirs = dist doc build .tox .eggs
 testpaths = test/
+# filter out deprecation warning spam coming from pytest_verbose_parametrize,
+# since we can't do anything about it until the plugin is updated
+filterwarnings =
```

Let's also make it stricter:
```suggestion
filterwarnings =
    error
```

We shouldn't be forgiving about other warnings, they should be fixed ASAP.

---

I like the idea on principle, but in practice, this is a sample of what happens in a `tox -e unit` run:
```
_ ERROR collecting test/unit/command/test_lint.py _
test/unit/command/test_lint.py:23: in <module>
    from molecule.command import lint
molecule/command/__init__.py:26: in <module>
    from molecule.command import cleanup  # noqa
molecule/command/cleanup.py:21: in <module>
    import click
.tox/unit/lib/python3.7/site-packages/click/__init__.py:18: in <module>
    from .core import Context, BaseCommand, Command, MultiCommand, Group, \
E     File "/home/semyers/ve/oasis/origin/molecule/.tox/unit/lib/python3.7/site-packages/click/core.py", line 1259
E       """
E        ^
E   SyntaxError: invalid escape sequence \s
```

Turning warning into errors means that unaddressed deprecations in dependencies will cause our tests to fail, which I think is unacceptable. For now, I'm going to limit the scope of change for the upcoming PR to only squashing the pytest_verbose_parametrize warning spam.

---

Fair enough so!